101-databricks-workspace-with-customer-managed-keys/prereqs/prereq.azuredeploy.json

```diff
@@ -0,0 +1,136 @@
+{
+  "$schema": "https://schema.management.azure.com/schemas/2019-04-01/deploymentTemplate.json#",
+  "contentVersion": "1.0.0.0",
+  "parameters": {
+    "workspaceName": {
+      "type": "string",
+      "metadata": {
+        "description": "The name of the Azure Databricks workspace to create."
+      }
+    },
+    "pricingTier": {
+      "type": "string",
+      "defaultValue": "premium",
+      "allowedValues": [
+        "standard",
+        "premium"
+      ],
+      "metadata": {
+        "description": "The pricing tier of workspace."
+      }
+    },
+    "location": {
+      "type": "string",
+      "defaultValue": "westus2",
+      "metadata": {
+        "description": "Location for all resources."
+      }
+    },
+    "keyVaultName": {
+      "type": "string",
+      "metadata": {
+        "description": "Specifies the name of the key vault."
+      }
+    },
+    "keyName": {
+      "type": "string",
+      "defaultValue": "adbTestKey",
+      "metadata": {
+        "description": "Azure Key Vault Key name."
+      }
+    },
+    "keyPermissions": {
+      "type": "array",
+      "metadata": {
+        "description": "Specifies the permissions to secrets in the vault. Valid values are: all, get, list, set, delete, backup, restore, recover, and purge."
+      }
+    }
+  },
+  "variables": {
+    "managedResourceGroupName": "[concat('databricks-rg-', parameters('workspaceName'), '-', uniqueString(parameters('workspaceName'), resourceGroup().id))]"
+  },
+  "resources": [
+    {
```

Comment:
why is the resource itself a prereq?

Reply:
Databricks workspace CMK has the following steps
1. Create workspace with prepareEncryption as true, this outputs MSI to be given access rights 
2. Once MSI emitted we need to configure KV access policies which I am doing in second step
3. After finishing these two prereqs, customer has to configure CMK at workspace which I am doing in main template